lib/experimental/class-wp-theme-json-resolver-gutenberg.php

```diff
@@ -33,7 +33,7 @@ public static function get_theme_data( $deprecated = array(), $settings = array(
 			_deprecated_argument( __METHOD__, '5.9' );
 		}
 
-		if ( null === static::$theme ) {
+		if ( null === static::$theme || ! static::$theme instanceof WP_Theme_JSON_Gutenberg ) {
```

Comment:
It might be good to add a comment with some instructions to remove the `|| ! static::$theme instanceof WP_Theme_JSON_Gutenberg` when backporting to core.

Reply:
Thanks, I'll add a comment.

Reply:
Added here: https://github.com/WordPress/gutenberg/pull/42756/commits/3673bc87fe550ebd0e20e939d8444d1cc385fb10